src/main/java/org/terasology/edibleFlora/genome/GenomeAuthoritySystem.java

```diff
@@ -0,0 +1,178 @@
+ */
+
+package org.terasology.edibleFlora.genome;
+
+import com.google.common.base.Function;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.genome.GenomeDefinition;
+import org.terasology.genome.GenomeRegistry;
+import org.terasology.genome.breed.BreedingAlgorithm;
+import org.terasology.genome.breed.ContinuousBreedingAlgorithm;
+import org.terasology.genome.breed.mutator.GeneMutator;
+import org.terasology.genome.breed.mutator.VocabularyGeneMutator;
+import org.terasology.genome.component.GenomeComponent;
+import org.terasology.genome.genomeMap.SeedBasedGenomeMap;
+import org.terasology.logic.characters.CharacterHeldItemComponent;
+import org.terasology.logic.common.ActivateEvent;
+import org.terasology.logic.common.RetainComponentsComponent;
+import org.terasology.logic.console.commandSystem.annotations.Command;
+import org.terasology.logic.console.commandSystem.annotations.Sender;
+import org.terasology.logic.delay.DelayedActionTriggeredEvent;
+import org.terasology.network.ClientComponent;
+import org.terasology.registry.In;
+import org.terasology.simpleFarming.components.BushDefinitionComponent;
+import org.terasology.simpleFarming.components.BushGrowthStage;
+import org.terasology.simpleFarming.components.SeedDefinitionComponent;
+import org.terasology.simpleFarming.events.AddGenomeRetention;
+import org.terasology.simpleFarming.events.BeforePlanted;
+import org.terasology.simpleFarming.events.ProduceCreated;
+import org.terasology.simpleFarming.events.TransferGenomeEvent;
+import org.terasology.simpleFarming.systems.PlantAuthoritySystem;
+import org.terasology.utilities.random.FastRandom;
+import org.terasology.world.WorldProvider;
+import org.terasology.world.block.BlockComponent;
+
+import javax.annotation.Nullable;
+
+/**
+ * System managing genetics of all plants
+ */
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class GenomeAuthoritySystem extends BaseComponentSystem {
+    @In
+    private GenomeRegistry genomeRegistry;
+    @In
+    private WorldProvider worldProvider;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GenomeAuthoritySystem.class);
+
+    /**
+     * Called immediately after a bush has been harvested.
+     * <p>
+     * Checks the genome component of the bush and assigns it to the seed if it had genes already.
+     * If the bush did not have a GenomeComponent, it is assigned as this is the first harvest
+     * A new Genome Definition is created for the family
+     *
+     * @param event the Produce created event
+     * @param creator the creator(bush) of the produce
+     */
+    @ReceiveEvent
+    public void onProduceCreated(ProduceCreated event, EntityRef creator) {
```

Comment:
So some elements of this could go into SF. In particular what I think would work better would be to store information about the specific crops possible genomic stuff as a component on its prefab. Then have SF read that and use the information there to mutate the crop as needed